django/db/models/sql/compiler.py

```diff
@@ -852,15 +856,17 @@ def execute_sql(self, result_type=MULTI):
             cursor, self.connection.features.empty_fetchmany_value,
             self.col_count
         )
-        if not self.connection.features.can_use_chunked_reads:
+        if not chunked_fetch and not self.connection.features.can_use_chunked_reads:
             try:
                 # If we are using non-chunked reads, we return the same data
                 # structure as normally, but ensure it is all read into memory
-                # before going any further.
+                # before going any further. If the user requests chunked_fetch
+                # we will honor that.
                 return list(result)
             finally:
-                # done with the cursor
-                cursor.close()
+                with self.connection.wrap_database_errors:
```

Comment:
I'm not sure if this change is necessary. This finally clause would only get executed with client-side cursors. In most places, the code does not wrap errors on `cursor.close()`, for example:
- https://github.com/francoisfreitag/django/blob/16614/server_side_cursor/django/contrib/gis/db/backends/postgis/introspection.py#L69
- https://github.com/francoisfreitag/django/blob/16614/server_side_cursor/django/contrib/gis/db/backends/spatialite/introspection.py#L60
- https://github.com/francoisfreitag/django/blob/16614/server_side_cursor/django/contrib/gis/db/backends/oracle/introspection.py#L48

I don't see why wrapping `cursor.close()` here would be necessary?